api/core/Directus/Util/StringUtils.php

```diff
@@ -120,13 +120,22 @@ public static function random($length = 16)
      * Random string shuffled from a list of alphanumeric characters
      *
      * @param int $length
+     * @param string $type
      *
      * @return string
      */
-    public static function randomString($length = 16)
+    public static function randomString($length = 16, $type = 'alphanumeric')
     {
         $pool = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
 
+        switch ($type) {
+            case 'numeric': $pool = '0123456789'; break;
+            case 'loweralpha': $pool = 'abcdefghijklmnopqrstuvwxyz'; break;
+            case 'upperalpha': $pool = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'; break;
+            case 'loweralphanumeric': $pool = '0123456789abcdefghijklmnopqrstuvwxyz'; break;
+            case 'upperalphanumeric': $pool = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ'; break;
```

Comment:
This can be improved to to concatenated the previous options.

Ex:

```php
$numericPool = '0123456789';
$upperAlphaPool = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
case 'upperalphanumeric': $pool = $numericPool . $upperAlpha; break;
```

Reply:
Yeh, at first I had some clever concat of numeric pool with alpha pool, and depending on upper or lower case just using php strtoupper() and strtolower, but this just seemed more straight forward for and simple.
```
$numeric = '0123456789';
$alpha = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
$pool = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
if ($type !== 'alphanumeric') {
  $pool = '';
  if (strpos($type, 'numeric') !== false) $pool .= $numeric;
  if (strpos($type, 'alpha') !== false) $pool .= $alpha;
  if (strpos($type, 'lower') !== false) $pool = strtolower($pool);
  if (strpos($type, 'upper') !== false) $pool = strtoupper($pool);
}
```

Reply:
Yeah, that's better you can make the $pool default to `$numeric . strtoupper() . strtolower()`.

All good, please update it so we can merge it! :)

Thanks!

Reply:
@WellingGuzman Is this change useful..?

It saves 30 characters, but decreases the readability of the code tremendously...

Reply:
done, tried to keep the switch statement to keep the code readable. @rijkvanzanten

Reply:
Perfect @u12206050. Thanks!